src/test/java/io/airlift/slice/TestSliceUtf8.java

```diff
@@ -116,6 +116,13 @@
         // min and max surrogate characters
         invalidSequences.add(new byte[] {(byte) 0b11101101, (byte) 0xA0, (byte) 0x80});
         invalidSequences.add(new byte[] {(byte) 0b11101101, (byte) 0xBF, (byte) 0xBF});
+
+        // check overlong encoding
+        invalidSequences.add(new byte[] {(byte) 0b11000000, (byte) 0xAF});
+        invalidSequences.add(new byte[] {(byte) 0b11100000, (byte) 0x90, (byte) 0x80});
```

Comment:
I would add a comment that says `// 3-byte encoding of 0x100`, and then also add a test case that tests 7FF.